internal/clients/nightfall/nightfall.go

```diff
@@ -278,3 +284,34 @@ func (n *Client) ReviewDiff(
 
 	return comments, nil
 }
+
+func filterFileDiffs(fileDiffs []*diffreviewer.FileDiff, fileIncludeList, fileExcludeList []string) []*diffreviewer.FileDiff {
+	filteredFileDiffs := fileDiffs
+	if fileIncludeList != nil && len(fileIncludeList) > 0 {
+		filteredFileDiffs = filterByFilePath(filteredFileDiffs, fileIncludeList, true)
+	}
+	if fileExcludeList != nil && len(fileExcludeList) > 0 {
+		filteredFileDiffs = filterByFilePath(filteredFileDiffs, fileExcludeList, false)
+	}
+	return filteredFileDiffs
+}
+
+func filterByFilePath(fileDiffs []*diffreviewer.FileDiff, regexPatterns []string, include bool) []*diffreviewer.FileDiff {
+	filteredFileDiffs := []*diffreviewer.FileDiff{}
+	for _, fd := range fileDiffs {
+		if (matchRegex(fd.PathNew, regexPatterns) && include) ||
+			(!matchRegex(fd.PathNew, regexPatterns) && !include) {
```

Comment:
I feel like we should make the call to `matchRegex` outside the if and store the result to be used in the if statement. Otherwise, I believe that we could potentially be making 2 calls to `matchRegex` with the same params.